signatures.md

```diff
@@ -52,163 +49,350 @@ Note: `Message` is used here as the object being signed, but this interface shou
+    
+}
+```
 
-## Wire Format
+## Key Generation & Elliptic Curve Parameters
 
-What bits are on the wire and in what order/format. We are currently adhering to libsecp256k1 serialization which is laid out as follows. Note that this format description may not be accurate. See the github link below for an authoritative format description.
+This specification assumes the implementer uses a strong cryptographic pseudorandom generator `RANDOM`. The following was adapted from [IETF BLS Signature Scheme - Keygen](https://tools.ietf.org/html/draft-boneh-bls-signature-00#section-2.2). 
 
-**Signature**
+```go
+Input: RANDOM
+Output: PK, SK
 
+   1.  SK = x, chosen as a random integer in the range 1 and r-1
+   2.  PK = x*P, P is an element of an elliptic curve E
+   3.  Output PK, SK
 ```
-sig SignatureBytes = [0x30][len][0x02][r][indicator][s][indicator][recovery]
+
+For ECDSA, `P` is the generator for `E`. For BLS, we choose `P` to be a generator in `E2` which is the larger of the elliptic curve groups.
+
+### BLS12-381 Parameter Choices
+
+We have chosen to use the same elliptic curve parameters as [zkcrypto/pairing](https://github.com/zkcrypto/pairing/tree/master/src/bls12_381) with the exception that we are signing in G1 and generating public keys in G2. Note: there is a namespace collision between [zkcrypto/pairing](https://github.com/zkcrypto/pairing/tree/master/src/bls12_381) and [IETF BLS Signature Scheme](https://tools.ietf.org/html/draft-boneh-bls-signature-00) concerning G1 and G2. In zkcrypto/paring, G1 and G2 are points defined in the largest prime order subgroup of the elliptic curve defined by BLS12-381. In the IETF BLS Signature Scheme, G1 and G2 represent the largest prime order subgroups in BLS12-381. We will comply with the IETF standard. We will denote the generators of G1 and G2 as P1 and P2 respectively. E1 and E2 will denote the elliptic curve defined by BLS12-381.
+
+```go
+q = 0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab
+
+r = 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001
+
+P1:
+x = 3685416753713387016781088315183077757961620795782546409894578378688607592378376318836054947676345821548104185464507
+y = 1339506544944476473020471379941921221584933875938349620426543736416511423956333506472724655353366534992391756441569
+
+P2:
+x = 3059144344244213709971259814753781636986470325476647558659373206291635324768958432433509563104347017837885763365758*u + 352701069587466618187139116011060144890029952792775240219908644239793785735715026873347600343865175952761926303160
+y = 927553665492332455747201965776037880757740193453592970025027978793976877002675564980949289727957565575433344219582*u + 1985150602287291935568054521177171638300868978215655730859378665066344726373823718423869104263333984641494340347905
+
 ```
 
-`s` = Scalar of size 32 bytes
+[TODO: add notes on point compression and effect on serialization once a standard has been set]
```

Comment:
Why not document the one from https://github.com/zkcrypto/pairing/tree/master/src/bls12_381#serialization for now, as that is what we are using?

Reply:
There is no standard way of doing this yet. I would prefer to include a standard once it is set instead of using the point compression in ZK-crypto and extra bits needed to indicate compression (3 bits for ZK-Crypto, I think?). If we just want to put something here that I can include that method for point compression and representation.

I have no particularly strong feelings on this as long as we properly indicate that we will adhere to the IETF standard (or companion) once it adds a method for point compression.

Reply:
Maybe just a link, to reference what we are doing in the meantime?

Reply:
Will add link.